api/router.js

```diff
@@ -0,0 +1,121 @@
+"use strict";
+
+const {query, db} = require("@arangodb");
+const createRouter = require("@arangodb/foxx/router");
+const joi = require("joi");
+const {generateBatchesCollection, generateBatchesGraph} = require("./dispatcher");
+const {
+    checkCollectionIsPresent,
+    checkGraphIsPresent,
+    sendInvalidInputMessage,
+    initialValidationGenerateEmbParams
+} = require("./validation");
+const {retrieveModel} = require("./model_metadata");
+const {metadataCollectionName, modelTypes} = require("../model/model_metadata");
+
+function initRouter() {
```

Comment:
FWIW you don't need to wrap this in a function. Routers are "inert" until they're mounted. You can just export the router directly if you want.

Reply:
Noted! I've modified that :)